### ISSUE TITLE
Upgrading to 6.x throws SyntaxError

### ISSUE THREAD

**Issue by btknorr** [auth0/node-jsonwebtoken#193]
```
const Hoek = require('hoek');
^^^^^
SyntaxError: Use of const in strict mode.
    at exports.runInThisContext (vm.js:73:16)
    at Module._compile (module.js:443:25)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/brianknorr/Dev/rd-express-server/node_modules/jsonwebtoken/sign.js:1:73)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
```

We are using node 0.12.9

Thanks,

Brian

**Comment by shakty** [comment]
Plus 1. Having the same issue.

**Comment by carlnordenfelt** [comment]
Same problem here, we downgraded to 5.7.0.

**Comment by amrendraw3** [comment]
This is actually an issue with the package dependencies. I have removed the folder (_**sudo rm -R jsonwebtoken**_) from my node_modules and installed the previous stable version of this package - (_**sudo npm install jsonwebtoken@5.6.0**_).

This works for me !!

Also change the same version in your _**package.json**_ file, if you used '*' for _`jsonwebtoken`_ module.

**Comment by carlnordenfelt** [comment]
@amrendraw3 Going back to a previous version is a workaround, not a solution to the issue. You are right in that, for most, the issue is caused by having jsonwebtoken: '*' in package.json (as in our case) but the actual issue lies in version 6.1.0. It might be intentional but in that case the issue has to be resolved as such.

PS. 5.7.0 was the last stable version prior to 6.*

**Comment by jfromaniello** [comment]
Thanks for pointing this out, we can try to find out a solution. This is not something in my code but in a dependency that I added: **Joi**.

However I don't know how long should I keep maintaining modules for node.js versions prior to v4, since v4 is the LTS and current stable version. In this case I don't see any other solution than implementing what I did with Joi in this package.

**Comment by carlnordenfelt** [comment]
@jfromaniello I think it's fine if version 6.* breaks compatibility with earlier node versions. It's a major update so there is no requirement to be backwards compatible. 
It would be good to point this out in the documentation though.

**Comment by btknorr** [comment]
Ya I agree...just document the compatibility and provide some sort of support for 5.7.  Or see if there is an older version of Joi that works with node 0.12.x and above :)

**Comment by justinffs** [comment]
@jfromaniello at a minimum you should include the node version dependancy in your package.json like joi does. 
````
"engines": {
    "node": ">=4.0.0"
  },
```
It would also be nice to have a change log and some sort of warning to users that node < 4 is not supported.

Also 0.12 is considered LTS and will be supported until next year https://github.com/nodejs/LTS

You could use an older version of joi that supports node < 4.  For now all of us need to use an old version of this library.

**Comment by jfromaniello** [comment]
Fixed in `v6.1.2` (I am using an older version of joi now).

**Comment by justinffs** [comment]
Awesome thank you!
